hls4ml/backends/fpga/passes/codegen.py

```diff
@@ -7,16 +7,27 @@ class GenerateConvIm2col(OptimizerPass):
     '''Generates tcode for im2col step of 1D/2d convolution'''
 
     def match(self, node):
-        return isinstance(node, (Conv1D, Conv2D)) and node.model.config.get_config_value('IOType') == 'io_parallel'
+        return (
+            isinstance(node, (Conv1D, Conv2D, SeparableConv1D, SeparableConv2D))
+            and node.model.config.get_config_value('IOType') == 'io_parallel'
+        )
 
     def transform(self, model, node):
-        node_class = node.__class__.__name__
-        if '1D' in node_class:
-            self._generate_im2col_1d(node)
-        elif '2D' in node_class:
-            self._generate_im2col_2d(node)
+        node_class = node.class_name
+        if 'Separable' in node_class:
```

Comment:
If we just had DepthwiseConv2D (https://keras.io/api/layers/convolution_layers/depthwise_convolution2d/) would this work? We nominally support it.

Reply:
It will work. For im2col algorithm the input matrix is the same, so DepthwiseConv1D/2D will be caught by the subsequent `elif` statement and from there on treated as the regular Conv1D/2D for creation of `fill_buffer` function. It is then correctly called in the new HLS. I've also added an explicit test for depthwise convolution